homeassistant/components/device_tracker/unifi.py

```diff
@@ -82,7 +90,8 @@ def _update(self):
             _LOGGER.error("Failed to scan clients: %s", ex)
             clients = []
 
```

Comment:
Probably looking for more something like:
`self._clients = {client['mac']: client for client in clients
                 if (dt_util.utcnow() - dt_util.utc_from_timestamp(float(
                     client['last_seen']))) < self._detection_time}`

So that you are actually setting _clients to only those seen within detection_time.